plots/generic.py

```diff
@@ -0,0 +1,539 @@
+    :param separatorargs: dict | [dict, dict] | None
+        If only one dict is given and groupingdepth == 2 the arguments are
+        applied to the separator of both level. Otherwise the arguments are
+        of separatorargs[0] are applied to the level 1 and [1] to level 2.
+        Arguments dictionary is passed on to matplotlib.pyplot.plot()
+        To turn of separators set it to None.
+    :param legend: boolean
+        Show legend?
+    :param legendargs: dict
+        Arguments dictionary is passed on to matplotlib.pyplot.legend()
+    :param ax: matplotlib axis | None (default)
+        The axis onto which to plot. If None a new figure is created.
+        When an axis is given, the function can't handle the figure settings.
+        Therefore it is recommended to call seaborn.set() with your preferred
+        settings before creating your matplotlib figure in order to control
+        your plotting layout.
+    :param style: str
+        seaborn style setting. Default: 'ticks'
+    :param palette: string | sequence
+        Define the color palette either by its name or use a custom palette in
+        a sequence of the form ([r,g,b],[r,g,b],...).
+    :param colorcodes: str
+        seaborn colorcodes setting. By default this is set to 'colorblind' so
+        that any additional drawing into the plot using colorcodes ('r', 'g', ...)
+        are coherent with the default color palette ('Set2') of the plot.
+    :param context: 'paper'(default) | 'talk' | 'poster'
+        seaborn context setting which controls the scaling of labels. For the
+        three options the parameters are scaled by .8, 1.3, and 1.6 respectively.
+    :return: ax, axhistx, axhisty   <matplotlib axis handle>
+        * ax is handle of the dot display plot
+        * axhistx is handle of the histogram plot above the the dot display
+        * axhisty is handle of the histogram plot on the right hand side
+    """
+
+    # Initialize plotting canvas
+    sns.set(style=style, palette=palette, context=context)
+    sns.set_color_codes(colorcodes)
+
+    if ax is None:
+        fig = plt.figure()
+        # axis must be created after sns.set() command for style to apply!
+        ax = fig.add_subplot(111)
+
+    margin = 1 - histscale
+    left, bottom, width, height = ax.get_position()._get_bounds()
+    ax.set_position([    left,                  bottom,
+                         margin * width,        margin * height])
+    axhistx = plt.axes([left,                   bottom + margin * height,
+                        margin * width,         histscale * height])
+    axhisty = plt.axes([left + margin * width,  bottom,
+                        histscale * width,      margin * height])
+
+    sns.despine(ax=axhistx)
+    sns.despine(ax=axhisty)
+
+    # Whitespace margin around dot display = 2%
+    ws_margin = 0.02
+
+    # Assertions
+    assert groupingdepth <= 2, "Grouping is limited to two layers."
```

Comment:
Please do not use `assert` to check user input (or other variable parameters). Rather use if ...: raise ValueError().

Asserts are used for debugging and should always pass. See, e.g., https://wiki.python.org/moin/UsingAssertionsEffectively

Asserts can be disabled:
http://stackoverflow.com/questions/1273211/disable-assertions-in-python